### PR TITLE
1898 Document CI Architecture

### PR DESCRIPTION
**Pull request by jonahx**
### What does this PR do?
- _What's changed? Why were these changes made?_
Adds an architectural diagram and detailed README explaining how our CI works.

### What ticket does this PR close?
#1898 

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [x] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [ ] This PR does not require updating any documentation

### CODE REVIEW

File: README_CI.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# Jenkins CI Architecture
+
+<!-- Note: This diagram can be re-regenerated by editing img/README_CI.puml.
+The VS Code PlantUML extension provides a good live-preview experience. -->
+![CI Architecture](./img/ci_architecture.svg)
```
Comment (jtuttle) [inline README_CI.md:5]: This is such a big enough diagram that it puts the ToC below the fold so I think I would prefer if the ToC came first.

File: README_CI.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
+# Jenkins CI Architecture
```
Comment (jtuttle) [inline README_CI.md:1]: This seems like it's more focused on just the test-running part of the CI. It doesn't talk about changelog validation or vulnerability scanning, for example. Would it be better to rename this something more specific or Is there more content to come?

**Review by jtuttle — CHANGES_REQUESTED** [review]

**Review by hughsaunders — APPROVED** [review]

File: README_CI.md (hunk 2 above)
Comment (jonahx) [reply README_CI.md:1]: I should probably add those.  Remember i said yesterday I was only ~80% done but wanted to ship :)

File: README_CI.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonahx) [reply README_CI.md:5]: I should probably make this clear, but this is "System" in the sense of the C4 model:

https://c4model.com/

(scroll down to "Abstractions")

That said, just re-reading those, it looks like it should be a "container".  I proabably have some other mis-classifications as well -- I'll review.

File: README_CI.md (hunk 2 above)
Comment (jtuttle) [reply README_CI.md:1]: Oh gotcha, so should I just review without them for now?

**Review by jtuttle — DISMISSED** [review]

**Review by jtuttle — APPROVED** [review]

**Pull request merged**